pandas/core/arrays/base.py

```diff
@@ -770,13 +795,13 @@ def shift(self, periods: int = 1, fill_value: object = None) -> ExtensionArray:
         )
         if periods > 0:
             a = empty
-            b = self[:-periods]
+            b: ExtensionArrayT = self[:-periods]
```

Comment:
this shouldn't be needed. `Revealed type is 'ExtensionArrayT`-1'` without the variable annotation. That's sort of the point of adding the overloads, so that the return types are known.

Reply:
`mypy` is complaining if I don't include that.  I opened up a `mypy` case and it's because we created the `ExtensionTypeT` type, which they say we don't need.  I'm going to experiment with that.

Reply:
> `mypy` is complaining if I don't include that

can you post the output. i'm not seeing that.

(what I am seeing is pylance saying the code is unreachable!)

Reply:
Solution suggested by mypy folks seems to be to avoid `ExtensionArrayT` as much as possible.  I've changed it to only use it twice in `pandas/core/arrays/base.py` .